### DISCUSSION TITLE
Add Specific Cat Token Lookup Functionality for Improved API Performance

### DISCUSSION THREAD

**Discussion by P-r-a-s-a-n-n-a** [spacescan-io/Spacescan-discussion#76] [Support]
Hey it would be great if we could specify a cat token to look up when making an api call. Currently when calling address/balance/(wallet)&type=cat if the wallet has a lot of different cats then it times out, so maybe being able to lookup just the one we want might help alleviate this?

**Comment by CheebaWizard** [comment]
Further from this, if the same would be possible with /address/transactions/{address} as well that would be awesome. :)

**Comment by nandhakumar2002** [comment]
@CheebaWizard Definitely the design is already there, it will be an enhancement of the address page
